### DISCUSSION TITLE
Better filters in user list

### DISCUSSION THREAD

**Discussion by buehler** [zitadel/zitadel#2221] [Ideas]
This correlates with https://github.com/caos/zitadel/discussions/2220.

It would be awesome to have better filter possibilities in the user list.
An "advanced" search that includes the filter of a user's state or additional data. Or even the possibility to search for duplicate email addresses.

Still the CSV export pointed to in 2220 would be a good addition to that matter.

We face the issue that some user that register themselves are doing so multiple times. Now it would be great to search for duplicates or status "initial" to see if there are any duplicates.

**Comment by livio-a** [comment]
Hi @buehler 

There's already the possibility to filter for the state in the user list:
```
message ListUsersRequest {
    //list limitations and ordering
    zitadel.v1.ListQuery query = 1;
    // the field the result is sorted
    zitadel.user.v1.UserFieldName sorting_column = 2;
    //criterias the client is looking for
    repeated zitadel.user.v1.SearchQuery queries = 3;
}
```
```
message SearchQuery {
    oneof query {
        option (validate.required) = true;
        ...
        StateQuery state_query = 7;
        ...
    }
}
```
REST representation:
```
{
	"queries": [
		{
			"state_query": {
				"state": "USER_STATE_INITIAL"
			}
		}
	]
}
```

I believe exporting a CSV and using excel's duplication detection would solve your current problem, right?
Or are there any other filters you would need directly in the api?

**Reply by buehler** [reply]
Effectively, exporting to CSV and using excel (or whatever) would solve this matter - yes.

Nevertheless, it would be a better user experience if the UI would have some better filter experience (simple / advanced search).
I'd imagine something like autoscout24.ch or something where one can search for basic stuff (like username, email, first/lastname) and some advanced filtering where all data can be searched for (status, last login, phone, has OTP/multifactor or not, etc)

**Reply by buehler** [reply]
sorry, read it wrong. The API does give (as I'm aware of) enough possibilities, but it should be reflected in the UI of zitadel.

**Reply by livio-a** [reply]
ok i agree :) i'll create an issue for the missing UI functionality

regarding extended / advanced search: this a topic for the reporting feature we're looking into later this year
therefore happy for any input
